packages/editor/src/editor.ts

```diff
@@ -0,0 +1,187 @@
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { once } from 'events';
+import { Readable } from 'stream';
+import spawn from 'cross-spawn';
+import { v4 as uuidv4 } from 'uuid';
+
+import type { ShellUserConfig, MongoshBus } from '@mongosh/types';
+import { signatures, ShellPlugin, ShellInternalState, TypeSignature } from '@mongosh/shell-api';
+
+export interface EditorOptions {
+  input: Readable;
+  vscodeDir: string,
+  tmpDir: string,
+  internalState: ShellInternalState;
+  makeMultilineJSIntoSingleLine: any;
+}
+
+export class Editor implements ShellPlugin {
+  _input: Readable;
+  _vscodeDir: string;
+  _tmpDir: string;
+  _tmpDocName: string;
+  _tmpDoc: string;
+  _internalState: ShellInternalState;
+  _makeMultilineJSIntoSingleLine: any;
+  load: (filename: string) => Promise<void>;
+  require: any;
+  config: { get<T extends keyof ShellUserConfig>(key: T): Promise<ShellUserConfig[T]> };
+  print: (...args: any[]) => Promise<void>;
+  npmArgv: string[];
+
+  constructor({ input, vscodeDir, tmpDir, internalState, makeMultilineJSIntoSingleLine }: EditorOptions) {
+    const { load, config, print, require } = internalState.context;
+    this._input = input;
+    this._vscodeDir = vscodeDir;
+    this._tmpDir = tmpDir;
+    this._tmpDocName = `edit-${uuidv4()}`;
+    this._tmpDoc = '';
+    this._internalState = internalState;
```

Comment:
This was renamed to `instanceState` in https://github.com/mongodb-js/mongosh/pull/1085 :)